store/RedisStore.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const noop = () => {}
+
+function RedisStore (store, timeWindow) {
+  this.store = store
+  this.timeWindow = timeWindow
+  this.key = 'fastify-rate-limit-'
+}
+
+RedisStore.prototype.incr = function (ip, cb) {
+  var key = this.key + ip
+  this.store.pipeline()
+    .incr(key)
+    .pttl(key)
+    .exec((err, result) => {
+      if (err) return cb(err)
```

Comment:
we do not want to bring down the full system if Redis goes down. Maybe we need to handle this error more gracefully? If Redis is down, we let everything go through?

Reply:
I don't know, personally I would block everything until redis is reachable again.
I've put a rate limiter for a reason, if I can't guarantee the limits I prefer that everything is blocked instead of leave the api unsafe.